File: loader/testdata/combat/game.lua
Game {
    title   = "Combat Test",
    author  = "Test",
    version = "0.1.0",
    start   = "cave",
    player_stats = {
        hp = 20, max_hp = 20,
        attack = 5, defense = 2,
    },
}

Room "cave" {
    description = "A dark cave.",
    exits = { south = "entrance" },
}

Room "entrance" {
    description = "The entrance.",
    exits = { north = "cave" },
}

Item "goblin_blade" {
    name        = "Rusty Goblin Blade",
    description = "A crudely forged blade.",
    location    = "cave",
    takeable    = true,
}

Enemy "cave_goblin" {
    name        = "Cave Goblin",
    description = "A snarling goblin clutching a rusty blade.",
    location    = "cave",
    stats = {
        hp = 12, max_hp = 12,
        attack = 4, defense = 1,
    },
    behavior = {
        { action = "attack", weight = 70 },
        { action = "defend", weight = 20 },
        { action = "flee",   weight = 10 },
    },
    loot = {
        items = { { id = "goblin_blade", chance = 50 } },
        gold  = 5,
    },
}

Rule("attack_goblin",
    When { verb = "attack", object = "cave_goblin" },
    Then {
        Say("You engage the Cave Goblin in combat!"),
        StartCombat("cave_goblin"),
    }
)

File: tools/editor/src/lib/codegen/game.ts
/**
 * Generate game.lua from GameMeta.
 */

import type { GameMeta } from '../model/export-types.js';
import { luaString } from './lua-utils.js';

export function generateGame(meta: GameMeta): string {
	const lines: string[] = [];
	lines.push('Game {');
	lines.push(`    title   = ${luaString(meta.title)},`);
	lines.push(`    author  = ${luaString(meta.author)},`);
	lines.push(`    version = ${luaString(meta.version)},`);
	lines.push(`    start   = ${luaString(meta.start)},`);
	lines.push(`    intro   = ${luaString(meta.intro)},`);

	if (meta.playerStats) {
		const s = meta.playerStats;
		lines.push('    player_stats = {');
		lines.push(`        hp = ${s.hp}, max_hp = ${s.maxHp},`);
		lines.push(`        attack = ${s.attack}, defense = ${s.defense},`);
		lines.push('    },');
	}

	lines.push('}');
	return lines.join('\n') + '\n';
}
